### ISSUE TITLE
Exporting/Downloading Files on Android

### ISSUE THREAD

**Issue by Splarkszter** [nspire-emus/firebird#337]
Hello. I tried to figure out how to export/download files to the main storage system but i can't seem to figure out how to do it.

There is no button like in the desktop version to browse the emulator file-system to download the file.

It would be much appreciated to have a way to do it. Thank you :)

**Comment by Vogtinator** [comment]
Indeed, currently there's no way except by transferring the flash image to a desktop system. It might be possible to implement file transfer in the other direction, by sending files from the calculator.

Implementing a full file tree view like the desktop version on mobile is not easy to do with QML and also really suitable for mobile platforms.

**Comment by Splarkszter** [comment]
Ok... Is there any cheap&easy way to do it? Like mounting the emulated filesystem to the phone storage? No UI would be needed from the APP side.

I don't understand how it works but Termux and RetroArch do something like that (show their internal file-system to the user trough the native "files" APP).

**Comment by Vogtinator** [comment]
No, there's no easy way to do it currently.

**Comment by UniversityDion** [comment]
Is there a way to export tns files from the calculator storage?. I want to export those tns files and import them onto my physical calculator. Really saves time in writing functions and things like that

**Comment by Vogtinator** [comment]
Only by transferring the flash image to a desktop version of firebird ATM.

**Comment by UniversityDion** [comment]
> Only by transferring the flash image to a desktop version of firebird ATM.

I am running firebird on my pc. Not very proficient with software. I can't seem to open the flash image. it says its file type is "File". how do i get around doing this?

**Comment by Vogtinator** [comment]
You need to set up firebird to use that flash image for booting.

**Comment by UniversityDion** [comment]
> You need to set up firebird to use that flash image for booting.

I am using that flash image on the emulator. Are there any options in the emulator to export the files that I make in them? 

Also what does the usb link function in the emulator do? I turned it on and nothing showed up. I tried to see if it would emulate a usb connection between my computer and the emulator but nothing showed up on the TI Computer Link software. If the connection can be established, I can get the tns files from there.

**Comment by Vogtinator** [comment]
> I am using that flash image on the emulator. Are there any options in the emulator to export the files that I make in them?

Yes.

> Also what does the usb link function in the emulator do? I turned it on and nothing showed up. I tried to see if it would emulate a usb connection between my computer and the emulator but nothing showed up on the TI Computer Link software. If the connection can be established, I can get the tns files from there.

You have to use the file explorer tab which you can select below the keypad. Click on the refresh symbol first, then you can download what you want.
